src/payment/strategies/paypal-commerce/paypal-commerce-payment-strategy.ts

```diff
@@ -55,22 +89,30 @@ export default class PaypalCommercePaymentStrategy implements PaymentStrategy {
         return Promise.reject(new OrderFinalizationNotRequiredError());
     }
 
-    deinitialize(): Promise<InternalCheckoutSelectors> {
-        this._paypalCommercePaymentProcessor.deinitialize();
+    async deinitialize(): Promise<InternalCheckoutSelectors> {
+        this._orderId = undefined;
 
         return Promise.resolve(this._store.getState());
     }
 
-    private async _getOrderId(methodId: string): Promise<string> {
-        const state = this._store.getState();
-        const cart = state.cart.getCartOrThrow();
-        const provider = methodId === 'paypalcommercecredit' ? 'paypalcommercecreditcheckout' : 'paypalcommercecheckout';
-        const { approveUrl, orderId } = await this._paypalCommerceRequestSender.setupPayment(provider, cart.id);
+    private _isPaypalCommerceOptionsPayments(options: PaypalCommercePaymentInitializeOptions | PaypalCommerceCreditCardPaymentInitializeOptions): options is PaypalCommercePaymentInitializeOptions {
+        return !!(options as PaypalCommercePaymentInitializeOptions).container;
```

Comment:
`return 'container' in options;` so you don't need to use `as` 🍹

Reply:
It doesn't work without `as`. I see 
`TS2339: Property 'container' does not exist on type 'PaypalCommerceInitializeOptions'.   Property 'container' does not exist on type 'PaypalCommerceCreditCardPaymentInitializeOptions'.`